src/react/components/TriggerEditor/index.tsx

```diff
@@ -0,0 +1,545 @@
+    const { value } = e.target;
+    setTriggerSource(value);
+    let triggerSourceJSON: Record<string, unknown> | null = null;
+    try {
+      triggerSourceJSON = JSON.parse(value);
+    } catch {
+      triggerSourceJSON = null;
+    }
+    if (!triggerSourceJSON) {
+      setTriggerSourceError('Invalid JSON');
+      return;
+    }
+    let trigger: AstarteTrigger | null = null;
+    try {
+      trigger = AstarteTrigger.fromJSON(triggerSourceJSON as any);
+    } catch {
+      trigger = null;
+    }
+    if (!trigger) {
+      setTriggerSourceError('Invalid Trigger');
+      return;
+    }
+    if (_.get(trigger, 'simpleTriggers[0].type') === 'data_trigger') {
+      handleFetchInterfacesForTrigger(trigger).then(setTriggerDraft);
+    } else {
+      setTriggerDraft(trigger);
+    }
+  }, []);
+
+  useEffect(() => {
+    const formattedTriggerSource = formatJSON(AstarteTrigger.toJSON(triggerDraft));
+    setTriggerSource((source) =>
+      formatJSONText(source) !== formattedTriggerSource ? formattedTriggerSource : source,
+    );
+    let validationErrors: { [key: string]: string } = {};
+    try {
+      AstarteTrigger.validation.validateSync(triggerDraft, {
+        abortEarly: false,
+        context: { realm, interface: triggerInterface },
+      });
+    } catch (err) {
+      validationErrors = _.mapValues(_.keyBy(_.uniqBy(err.inner, 'path'), 'path'), 'message');
+    }
+    setTriggerValidationErrors(validationErrors);
+    setTriggerSourceError(Object.values(validationErrors).join('. '));
+    if (onChange) {
+      const isValidTrigger = _.isEmpty(validationErrors);
+      onChange(triggerDraft, isValidTrigger);
+    }
+  }, [triggerDraft, realm, triggerInterface]);
+
+  useEffect(() => {
+    if (initialData) {
+      handleFetchInterfacesForTrigger(initialData);
+    } else {
+      handleFetchInterfacesName();
+    }
+  }, []);
+
+  return (
+    <Row>
+      <Col md={isSourceVisible ? 6 : 12}>
+        <Container fluid className="bg-white rounded p-3">
+          <Form>
+            <Form.Row className="mb-2">
+              <Col sm={12}>
+                <Form.Group controlId="triggerName">
+                  <Form.Label>Name</Form.Label>
+                  <Form.Control
+                    type="text"
+                    autoComplete="off"
+                    required
+                    readOnly={isReadOnly}
+                    value={_.get(triggerDraft, 'name')}
+                    onChange={handleTriggerNameChange}
+                    isInvalid={_.get(triggerValidationErrors, 'name') != null}
+                  />
+                  <Form.Control.Feedback type="invalid">
+                    {_.get(triggerValidationErrors, 'name')}
+                  </Form.Control.Feedback>
+                </Form.Group>
+              </Col>
+            </Form.Row>
+          </Form>
+          <SimpleTriggerForm
+            fetchInterface={handleFetchInterface}
+            fetchInterfacesName={handleFetchInterfacesName}
```

Comment:
this is only done once as soon as you open the page. You can fetch the list of interfaces in the main component, and pass it down as a property.

Reply:
Nice catch, I fixed it. Now the list of interfaces is passed down as a property, no re-fetching needed.